scale_build/packages/git.py

```diff
@@ -117,6 +123,43 @@ def checkout(self, branch_override=None, retries=3):
         log = 'Checkout ' if not update else 'Updating '
         logger.info(log + 'of git repo %r (using branch %r) complete', self.name, branch)
 
+    @property
+    def git_args(self):
+        if self.ssh_based_source:
+            return [
+                'git', '-c',
+                f'core.sshCommand=ssh -i {self.get_identity_file_path} -o StrictHostKeyChecking=\'accept-new\''
+            ]
+        else:
+            return ['git']
+
+    @property
+    def get_identity_file_path(self):
```

Comment:
We should enforce that an absolute path is given to us. `os.path.abspath` doesn't work, for example, if user sets something like `~/.ssh/id_rsa` (I just hit this problem testing this branch 😄 )

Reply:
This should cater to that https://github.com/truenas/scale-build/pull/323/commits/bdc665586a4418053090852db1fd6d630144afeb. Let me know if you spot any issues on that end please 🤔 

I just think it is convenient to be able to put any custom keys in `identities/` and then specify an env variable `./identities/abckey` from CI perspective as this way the builder can live anywhere on the host